modules/storages/spec/factories/storage_factory.rb

```diff
@@ -29,16 +29,22 @@
 #++
 
 FactoryBot.define do
-  factory :storage, class: '::Storages::Storage' do
+  factory :storage, class: 'Storages::Storage' do
     sequence(:name) { |n| "Storage #{n}" }
     creator factory: :user
 
+    # rubocop:disable FactoryBot/FactoryAssociationWithStrategy
+    # For some reason the order of saving breaks STI
     trait :with_oauth_client do
-      oauth_client
+      oauth_client { build(:oauth_client) }
     end
+    # rubocop:enable FactoryBot/FactoryAssociationWithStrategy
 
-    factory :nextcloud_storage, class: '::Storages::NextcloudStorage' do
-      provider_type { Storages::Storage::PROVIDER_TYPE_NEXTCLOUD }
```

Comment:
🔴 ❓  Why is this no longer needed? It had the effect, that e.g. running `create(:storage)` inside a spec would return an error, as you cannot instantiate a storage without the mandatory attribute of `provider_type`. 
As it is removed from the sub factories, I believe, this check will no longer work? Or does it?

Reply:
`provider_type` will get automatically filled in when the record is saved as RoR STI mechanism kicks in. So this wasn't needed unless we wanted to use the `:storage` factory -  which we shouldn't as, we discussed previously - it is basically an `abstract class`.

Reply:
Ah, it it still works. cool! Then no change requested. 🟢